gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -806,40 +806,12 @@ public void write(JsonWriter out, Locale value) throws IOException {
 
   public static final TypeAdapterFactory LOCALE_FACTORY = newFactory(Locale.class, LOCALE);
 
-  /*
-   * The following adapter and factory fields have not been removed yet and are only deprecated
-   * for now because external projects might be using them, despite being part of Gson's internal
-   * implementation.
-   */
-
-  /**
-   * @deprecated {@code TypeAdapters} is an internal Gson class. To obtain the adapter for {@link
-   *     JsonElement} and subclasses use instead:
-   *     <pre>{@code
-   * TypeAdapter<JsonElement> adapter = gson.getAdapter(JsonElement.class);
-   * }</pre>
-   */
-  @Deprecated
   public static final TypeAdapter<JsonElement> JSON_ELEMENT = JsonElementTypeAdapter.ADAPTER;
 
-  /**
-   * @deprecated {@code TypeAdapters} is an internal Gson class. To obtain the adapter for {@link
-   *     JsonElement} and subclasses use instead:
-   *     <pre>{@code
-   * TypeAdapter<JsonElement> adapter = gson.getAdapter(JsonElement.class);
-   * }</pre>
-   */
-  @Deprecated
-  public static final TypeAdapterFactory JSON_ELEMENT_FACTORY = JsonElementTypeAdapter.FACTORY;
+  public static final TypeAdapterFactory JSON_ELEMENT_FACTORY =
+      TypeAdapters.newTypeHierarchyFactory(JsonElement.class, JSON_ELEMENT);
```

Comment:
Really minor: You can omit the `TypeAdapters.` qualifier here (unless you added it intentionally) since this is already within `TypeAdapters`.
```suggestion
      newTypeHierarchyFactory(JsonElement.class, JSON_ELEMENT);
```